exercises/variable-length-quantity/.meta/gen.go

```diff
@@ -0,0 +1,146 @@
+// +build ignore
+
+package main
+
+import (
+	"log"
+	"text/template"
+
+	"../../../gen"
+)
+
+func main() {
+	t := template.New("").Funcs(template.FuncMap{
+		"GroupComment": GroupComment,
+		"iByteSlice":   iByteSlice,
+		"byteSlice":    byteSlice,
+		"iUint32Slice": iUint32Slice,
+		"lengthSlice":  lengthSlice,
+	})
+	t, err := t.Parse(tmpl)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var j js
+	if err := gen.Gen("variable-length-quantity", &j, t); err != nil {
+		log.Fatal(err)
+	}
+}
+
+// iByteSlice converts a slice of interface{} values whose
+// underlying members should be float64(JSON decoded integers)
+// to a slice of byte.
+func iByteSlice(ns []interface{}) []byte {
+	b := make([]byte, len(ns))
+	for i, n := range ns {
+		if fn, ok := n.(float64); ok {
+			b[i] = byte(int(fn))
+		}
+	}
+	return b
+}
+
+// byteSlice converts a slice of uint32 to a byte slice.
+func byteSlice(ns []uint32) []byte {
+	b := make([]byte, len(ns))
+	for i, n := range ns {
+		b[i] = byte(n)
+	}
+	return b
+}
+
+// lengthSlice returns the length of given slice.
+func lengthSlice(ns []uint32) int {
+	return len(ns)
+}
+
+// iUint32Slice converts an interface{} to a slice
+// of uint32 values or an empty slice.
+func iUint32Slice(ns interface{}) []uint32 {
+	is, ok := ns.([]interface{})
+	if !ok {
+		return []uint32{}
+	}
+	u := make([]uint32, len(is))
+	for i, n := range is {
+		if un, ok := n.(float64); ok {
+			u[i] = uint32(un)
+		}
+	}
+	return u
+}
+
+// The JSON structure we expect to be able to unmarshal into
+type js struct {
+	Groups []TestGroup `json:"Cases"`
+}
+
+type TestGroup struct {
+	Description string
+	Cases       []OneCase
+}
+
+type OneCase struct {
+	Description string
+	Property    string      // "encode" or "decode"
+	Input       []uint32    // supports both []byte and []uint32 in JSON.
+	Expected    interface{} // supports []byte, []uint32, or null in JSON.
```

Comment:
two thing stand out to me:

1. I was a bit surprised: I thought that since `nil` is assignable to `[]uint32` that `uint32` can be used here
2. I feel like the schema is not doing well for us. In the old way, where `encode` and `decode` are keys, we can have a different struct for each kind of case. But since now we can only tell by looking at `property`, we have to make our `OneCase` struct a union of all possible cases. Since you have written more generators than I have, I would ask you: How much does this bother you?